docs/azureai/azureai-search-document-integration.md

```diff
@@ -71,80 +204,70 @@ public class ExampleService(SearchIndexClient indexClient)
-
-var myService = builder.AddProject<Projects.MyService>()
-                       .WithReference(search);
+public class ExampleService(
+    [KeyedService("images")] SearchIndexClient imagesClient,
+    [KeyedService("documents")] SearchIndexClient documentsClient)
+{
+    // Use clients...
+}
 ```
 
-The <xref:Aspire.Hosting.AzureSearchExtensions.AddAzureSearch%2A> method will read connection information from the AppHost's configuration (for example, from "user secrets") under the `ConnectionStrings:search` config key. The `WithReference` method passes that connection information into a connection string named `search` in the `MyService` project. In the _:::no-loc text="Program.cs":::_ file of `MyService`, the connection can be consumed using:
+For more information, see [Keyed services in .NET](/dotnet/core/extensions/dependency-injection#keyed-services).
 
-```csharp
-builder.AddAzureSearch("search");
-```
-
-## Configuration
+### Configuration
 
-The .NET Aspire Azure Azure Search library provides multiple options to configure the Azure Search Service based on the requirements and conventions of your project. Note that either an `Endpoint` or a `ConnectionString` is required to be supplied.
+The .NET Aspire Azure AI Search Documents library provides multiple options to configure the Azure AI Search connection based on the requirements and conventions of your project. Either an `Endpoint` or a `ConnectionString` is required to be supplied.
 
-### Use a connection string
+#### Use a connection string
 
-A connection can be constructed from the **Keys and Endpoint** tab with the format `Endpoint={endpoint};Key={key};`. You can provide the name of the connection string when calling `builder.AddAzureSearch()`:
+A connection can be constructed from the **Keys and Endpoint** tab with the format `Endpoint={endpoint};Key={key};`. You can provide the name of the connection string when calling `builder.AddAzureSearchClient()`:
 
 ```csharp
-builder.AddAzureSearch("searchConnectionName");
+builder.AddAzureSearchClient("searchConnectionName");
 ```
 
-And then the connection string will be retrieved from the `ConnectionStrings` configuration section. Two connection formats are supported:
+The connection string is retrieved from the `ConnectionStrings` configuration section. Two connection formats are supported:
 
-#### Account endpoint
+##### Account endpoint
 
 The recommended approach is to use an `Endpoint`, which works with the `AzureSearchSettings.Credential` property to establish a connection. If no credential is configured, the <xref:Azure.Identity.DefaultAzureCredential> is used.
 
 ```json
 {
   "ConnectionStrings": {
-    "searchConnectionName": "https://{search_service}.search.windows.net/"
+    "search": "https://{search_service}.search.windows.net/"
   }
 }
 ```
 
-#### Connection string
+##### Connection string
 
-Alternatively, a custom connection string can be used.
+Alternatively, a custom connection string can be used:
```

Comment:
nit: If this is the non-recommended approach, it might be good to say something brief about how the other one is more secure or something.